tests/testthat/test-Model-methods.R

```diff
@@ -21,6 +21,30 @@ test_that("doseFunction-GeneralModel returns correct dose function", {
   expect_identical(dose_fun_env[["samples"]], samples)
 })
 
+test_that("doseFunction-GeneralModel returns correct dose function for matrix param", {
+  model <- h_get_logistic_log_normal_mix()
+  params <- list(
+    alpha0 = matrix(c(-0.94, -0.94, -2.37, -2.37, -0.67, -0.67, -1.28, -1.08), nrow = 4),
+    alpha1 = matrix(c(0.45, 0.45, 0.40, 0.40, 0.75, 0.75, 1.18, 0.63), nrow = 4),
+    comp = c(1, 1, 1, 1)
+  )
+  samples <- h_as_samples(params, burnin = 10000, fixed = FALSE)
+  dose_args <- c("x", "model", "samples")
+
+  dose_fun <- doseFunction(model, alpha0 = params$alpha0, alpha1 = params$alpha1, comp = params$comp)
+  dose_fun_dose_args <- as.character(body(dose_fun)[[2]][-1])
+  dose_fun_env <- environment(dose_fun)
+
+  expect_function(dose_fun, args = "x", nargs = 1, null.ok = FALSE)
+  expect_equal(dose_fun_dose_args, dose_args)
```

Comment:
`dose_args` is only used here - better readable if you avoid that variable and just put the character vector here
```suggestion
  expect_equal(dose_fun_dose_args, c("x", "model", "samples"))
```